pkg/azuredisk/azuredisk_v2.go

```diff
@@ -110,60 +114,121 @@ type DriverV2 struct {
+	return newDriverV2(config)
+}
+
+func getDriverConfig(driverConfig *azdiskv1beta2.AzDiskDriverConfiguration) {
+	if *driverConfigPath != "" {
+		if driverConfig.ObjectNamespace == "" {
+			driverConfig.ObjectNamespace = consts.DefaultAzureDiskCrdNamespace
+		}
+		if driverConfig.ControllerConfig.Enabled == nil {
+			c := consts.DefaultIsControllerPlugin
+			driverConfig.ControllerConfig.Enabled = &c
+		}
+		if driverConfig.ControllerConfig.LeaseDurationInSec == nil {
+			c := consts.DefaultControllerLeaseDurationInSec
+			driverConfig.ControllerConfig.LeaseDurationInSec = &c
+		}
+		if driverConfig.ControllerConfig.LeaseRenewDeadlineInSec == nil {
+			c := consts.DefaultControllerLeaseRenewDeadlineInSec
+			driverConfig.ControllerConfig.LeaseRenewDeadlineInSec = &c
+		}
+		if driverConfig.ControllerConfig.LeaseRetryPeriodInSec == nil {
+			c := consts.DefaultControllerLeaseRetryPeriodInSec
+			driverConfig.ControllerConfig.LeaseRetryPeriodInSec = &c
+		}
+		if driverConfig.ControllerConfig.LeaderElectionNamespace == "" {
+			driverConfig.ControllerConfig.LeaderElectionNamespace = consts.ReleaseNamespace
+		}
+		if driverConfig.ControllerConfig.PartitionName == "" {
+			driverConfig.ControllerConfig.PartitionName = consts.DefaultControllerPartitionName
+		}
+		if driverConfig.NodeConfig.Enabled == nil {
+			c := consts.DefaultIsNodePlugin
+			driverConfig.NodeConfig.Enabled = &c
+		}
+		if driverConfig.NodeConfig.HeartbeatFrequencyInSec == nil {
+			c := consts.DefaultHeartbeatFrequencyInSec
+			driverConfig.NodeConfig.HeartbeatFrequencyInSec = &c
+		}
+		if driverConfig.NodeConfig.PartitionName == "" {
+			driverConfig.NodeConfig.PartitionName = consts.DefaultNodePartitionName
+		}
+		// Emit warning log for using deprecated command-line parameters
+		flag.Visit(func(f *flag.Flag) {
+			if slices.Contains(consts.CommandLineParams, f.Name) {
+				klog.Warningf("the command-line parameter %v is deprecated and overridden by CongfigMap", f.Name)
+			}
+		})
+	} else {
+		driverConfig.ObjectNamespace = *driverObjectNamespace
+		driverConfig.ControllerConfig.Enabled = isControllerPlugin
+		driverConfig.ControllerConfig.LeaseDurationInSec = controllerLeaseDurationInSec
+		driverConfig.ControllerConfig.LeaseRenewDeadlineInSec = controllerLeaseRenewDeadlineInSec
+		driverConfig.ControllerConfig.LeaseRetryPeriodInSec = controllerLeaseRetryPeriodInSec
+		driverConfig.ControllerConfig.LeaderElectionNamespace = *leaderElectionNamespace
+		driverConfig.ControllerConfig.PartitionName = *controllerPartition
+		driverConfig.NodeConfig.Enabled = isNodePlugin
+		driverConfig.NodeConfig.HeartbeatFrequencyInSec = heartbeatFrequencyInSec
+		driverConfig.NodeConfig.PartitionName = *nodePartition
+		// Emit warning log for using deprecated command-line parameters
+		flag.Visit(func(f *flag.Flag) {
+			if slices.Contains(consts.CommandLineParams, f.Name) {
+				klog.Warningf("the command-line parameter %v is deprecated, setting by CongfigMap is supported", f.Name)
```

Comment:
Does deprecating the CLI parameters also means we will remove them eventually?

Reply:
@bertinatto Yes, that is correct. We'll leave them in for a few releases though and clearly communicate when we remove them.